vsifile/io/base.py

```diff
@@ -63,30 +63,37 @@ class BaseReader(metaclass=abc.ABCMeta):
     header_cache: Cache = field(init=False, factory=lambda: header_cache)
 
     _key: str = field(init=False)
-    _store: ObjectStore = field(init=False)
     _loc: int = field(init=False, default=0)
     _closed: bool = field(init=False, default=True)
     _size: int = field(init=False)
     _mtime: datetime.datetime = field(init=False)
     _seekable: bool = field(init=False, default=False)
 
-    @abc.abstractmethod
+    def __attrs_post_init__(self):
+        """init class."""
+        self.store = self.store or obs.store.from_url(
+            self.name,
+            config=self.config,
+            client_options=self.client_options,
+            retry_config=self.retry_config,
+        )
```

Comment:
`obs.store.from_url` does not guaranty to work all the time, which is why we have our own wrapper over each store (http, file and s3) 

e.g https://github.com/developmentseed/obstore/issues/321